### PR TITLE
Status descriptions on login

### PR DESCRIPTION
**Pull request by MaximilianMeiler**
May need a css revamp...just like the entire rest of the login page

### CODE REVIEW

File: Frontend/src/Login.jsx [hunk 1]
```diff
@@ -41,7 +45,8 @@ export default function Login({apiKey, setApiKey, setUserData, setUserId, setCou
 
         
       } catch (e) {
-        console.log("login failed");
+        console.log(e.response.data.message);
+        setLoginResponse(e.response.data.message);
```
Comment (kthchew) [inline Frontend/src/Login.jsx:49]: this doesn't check if there's actually a response (a backend that is down or times out will cause the front end to simply stall on logging in forever)

**Review by kthchew — CHANGES_REQUESTED** [review]
cool

**Review by MaximilianMeiler — COMMENTED** [review]
@kthchew Can you attempt to continue this fix? I think you're more familiar with the backend's error structure than I am

**Comment by kthchew** [comment]
@MaximilianMeiler I think this should fix it. I changed the errors from `canvas.js` so they should contain a recommended status code as well to make things easier. I haven't been able to trigger the rate limit but I tried invalid API keys and they didn't crash the server.

For reference, essentially any exported function in `canvas.js` (that contacts the Canvas API) can throw. It needs to be caught when used like so:

```js
try {
  const user = await canvas.getUser(canvas_api_token);
  return res.json(user);
} catch (error) {
  return res.status(error.status).json({ error: error.message });
}
```

**Review by kthchew — APPROVED** [review]
lgtm

**Pull request merged**